bids/cli.py

```diff
@@ -0,0 +1,123 @@
+from pathlib import Path
+import click
+
+from . import __version__
+
+# alias -h to trigger help message
+CONTEXT_SETTINGS = {'help_option_names': ['-h', '--help']}
+
+
+class PathOrRegex(click.ParamType):
+    "A helper Type to parse BIDSLayoutIndexer ignore/force entries"
+    name = "path or /regex/"
+
+    def convert(self, value, param, ctx):
+        if value.startswith('/') and value.endswith('/'):
+            # regex pattern
+            import re
+            value = re.compile(value[1:-1])
+        # otherwise, return as is
+        return value
+
+
+# create group of commands as entrypoint
+@click.group(context_settings=CONTEXT_SETTINGS)
+@click.version_option(__version__, prog_name='pybids')
+def cli():
+    """Command-line interface for PyBIDS operations"""
+    pass
+
+
+@cli.command(context_settings=CONTEXT_SETTINGS)
+@click.argument('root', type=click.Path(file_okay=False, exists=True))
+@click.option('--db-path', type=click.Path(file_okay=False, resolve_path=True),
+              help="Path to save database index.")
+@click.option('--derivatives', multiple=True, default=False, show_default=True,
```

Comment:
If I just want `derivatives=True`, how do I specify that? I would expect just `--derivatives`, but the semantics here are unclear and it's not tested.

Reply:
good catch, I didn't check for this.